promoter/database.go

```diff
@@ -0,0 +1,132 @@
+package promoter
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"go.sia.tech/siad/crypto"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readconcern"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+	"go.mongodb.org/mongo-driver/mongo/writeconcern"
+)
+
+const (
+	dbName                  = "siacoinpromoter"
+	colWatchedAddressesName = "watchedaddresses"
+)
+
+type (
+	// updateFunc is the type of a function that can be used as a callback
+	// in threadedAddressWatcher.
+	updateFunc func(WatchedAddressesUpdate)
+
+	// WatchedAddress describes an entry in the watched address collection.
+	WatchedAddress struct {
+		// Address is the actual address we track. We make that the _id
+		// of the object since the addresses should be indexed and
+		// unique anyway.
+		Address crypto.Hash `bson:"_id"`
+	}
+
+	// WatchedAddressesUpdate describes an update to the watched address
+	// collection.
+	WatchedAddressesUpdate struct {
+		DocumentKey struct {
+			Address crypto.Hash `bson:"_id"`
+		} `bson:"documentKey"`
+		OperationType string `bson:"operationType"`
```

Comment:
Our current convention is snake_case for field names. `_id` already starts with an underscore.

Reply:
This is not a field I'm creating. It's the convention used by mongo itself.